src/ucp/wireup/wireup.c

```diff
@@ -1052,65 +1052,23 @@ ucp_wireup_check_config_intersect(ucp_ep_h ep,
     uct_ep_h new_uct_eps[UCP_MAX_LANES]            = { NULL };
     ucp_ep_config_key_t *old_key;
     ucp_lane_index_t lane, reuse_lane;
-    ucp_wireup_ep_t *cm_wireup_ep;
 
     ucs_queue_head_init(replay_pending_queue);
 
     if ((ep->cfg_index == UCP_WORKER_CFG_INDEX_NULL) ||
-        !ucp_ep_has_cm_lane(ep) ||
-        !ucp_ep_is_sockaddr_stub(ep)) {
+        !ucp_ep_has_cm_lane(ep) || !ucp_ep_is_sockaddr_stub(ep)) {
         return;
     }
 
-#if UCS_ENABLE_ASSERT
     for (lane = 0; lane < ucp_ep_num_lanes(ep); ++lane) {
         ucs_assert(ep->uct_eps[lane] != NULL);
     }
-#endif
 
     old_key = &ucp_ep_config(ep)->key;
 
     ucp_ep_config_lanes_intersect(old_key, new_key, reuse_lane_map);
 
-    if (ucp_ep_config_key_has_cm_lane(new_key)) {
-        /* CM lane has to be re-used by the new EP configuration */
-        ucs_assert(reuse_lane_map[ucp_ep_get_cm_lane(ep)] != UCP_NULL_LANE);
-        ucs_assert(old_key->wireup_lane != UCP_NULL_LANE);
-
-        /* set the correct WIREUP MSG lane in case of CM */
-        if (new_key->wireup_lane != UCP_NULL_LANE) {
-            reuse_lane = ucp_ep_config_find_lane_index(old_key,
-                                                       new_key->wireup_lane,
-                                                       reuse_lane_map);
-            if (reuse_lane != UCP_NULL_LANE) {
-                new_uct_eps[new_key->wireup_lane] = ep->uct_eps[reuse_lane];
-                /* reset the UCT EP from the previous lane configuration to not
-                 * destroy it, since it will be used by the new configuration */
-                ep->uct_eps[reuse_lane]           = NULL;
-            }
-        } else {
-            reuse_lane = reuse_lane_map[old_key->wireup_lane];
-            if (reuse_lane != UCP_NULL_LANE) {
-                new_key->wireup_lane              = reuse_lane;
-            } else /* old wireup lane won't be re-used */ {
-                cm_wireup_ep                      = ucp_ep_get_cm_wireup_ep(ep);
-                new_key->wireup_lane              = new_key->cm_lane;
-                reuse_lane                        = old_key->wireup_lane;
-                /* save UCT EP from the previous WIREUP lane to the CM WIREUP
-                 * auxiliary EP */
-                cm_wireup_ep->aux_ep              = ep->uct_eps[reuse_lane];
-                cm_wireup_ep->aux_rsc_index       = old_key->lanes[reuse_lane].rsc_index;
-                /* reset the UCT EP from the previous WIREUP lane to not
-                 * destroy it, since it's not needed anymore in the new
-                 * configuration, but will be used for WIREUP MSG */
-                ep->uct_eps[reuse_lane]           = NULL;
-            }
-        }
-    }
-
-    if (!ucp_ep_has_cm_lane(ep) &&
-        ucp_ep_is_sockaddr_stub(ep) &&
-        (ep->uct_eps[0] != NULL)) {
+    if (ucp_ep_is_sockaddr_stub(ep) && (ep->uct_eps[0] != NULL)) {
```

Comment:
how can ucp_ep_is_sockaddr_stub be true, if ucp_ep_has_cm_lane is false (line 1059)?